govcd/org.go

```diff
@@ -5,74 +5,391 @@
+		}
+
+	}
+
+	return nil
+}
+
+// Removes All networks in the org
+func (adminOrg *AdminOrg) removeAllOrgNetworks() error {
+	for _, a := range adminOrg.AdminOrg.Networks.Networks {
+		// Get Network HREF
+		networkHREF := adminOrg.c.VCDHREF
+		networkHREF.Path += "/admin/network/" + strings.Split(a.HREF, "/network/")[1] //gets id
+		req := adminOrg.c.NewRequest(map[string]string{}, "DELETE", networkHREF, nil)
+		resp, err := checkResp(adminOrg.c.Http.Do(req))
+		if err != nil {
+			return fmt.Errorf("error deleting newtork: %s, %s", err, networkHREF.Path)
+		}
+
+		task := NewTask(adminOrg.c)
+		if err = decodeBody(resp, task.Task); err != nil {
+			return fmt.Errorf("error decoding task response: %s", err)
+		}
+		if task.Task.Status == "error" {
+			return fmt.Errorf("network not properly destroyed")
+		}
+		err = task.WaitTaskCompletion()
+		if err != nil {
+			return fmt.Errorf("Couldn't finish removing network %#v", err)
+		}
+	}
+	return nil
+}
+
+// Forced removal of all organization catalogs
+func (adminOrg *AdminOrg) removeCatalogs() error {
+	for _, a := range adminOrg.AdminOrg.Catalogs.Catalog {
+		// Get Catalog HREF
+		catalogHREF := adminOrg.c.VCDHREF
+		catalogHREF.Path += "/admin/catalog/" + strings.Split(a.HREF, "/catalog/")[1] //gets id
+		req := adminOrg.c.NewRequest(map[string]string{
+			"force":     "true",
+			"recursive": "true",
+		}, "DELETE", catalogHREF, nil)
+		_, err := checkResp(adminOrg.c.Http.Do(req))
+		if err != nil {
+			return fmt.Errorf("error deleting catalog: %s, %s", err, catalogHREF.Path)
+		}
+	}
+	return nil
+
+}
+
+// Given a valid catalog name, FindCatalog returns a Catalog object. Otherwise
+// it returns an error. This is the adminOrg version of FindCatalog, but it does 
+// NOT return an admin version of a Catalog.
+func (adminOrg *AdminOrg) FindCatalog(catalogName string) (Catalog, error) {
+	for _, a := range adminOrg.AdminOrg.Catalogs.Catalog {
+		// Get Catalog HREF
+		if a.Name == catalogName {
+			splitbyAdminHREF := strings.Split(a.HREF, "/admin")
+			catalogHREF := splitbyAdminHREF[0] + splitbyAdminHREF[1]
+			catalogURL, err := url.ParseRequestURI(catalogHREF)
+			if err != nil {
+				return Catalog{}, fmt.Errorf("error decoding catalog url: %s", err)
+			}
+			req := adminOrg.c.NewRequest(map[string]string{}, "GET", *catalogURL, nil)
+			resp, err := checkResp(adminOrg.c.Http.Do(req))
+			if err != nil {
+				return Catalog{}, fmt.Errorf("error retreiving catalog: %s", err)
+			}
+			cat := NewCatalog(adminOrg.c)
+
+			if err = decodeBody(resp, cat.Catalog); err != nil {
+				return Catalog{}, fmt.Errorf("error decoding catalog response: %s", err)
+			}
+
+			// The request was successful
+			return *cat, nil
+		}
+	}
+	return Catalog{}, fmt.Errorf("can't find catalog: %s", catalog)
 }
 
-func (o *Org) FindCatalog(catalog string) (Catalog, error) {
+// Given a valid catalog name, FindCatalog returns a Catalog object. Otherwise
+// it returns an error.
```

Comment:
So Im not sure what the right way to go about this as all the get methods in the sdk work that way. However I don't think it should be an issue because issue #58 will make it so that an client can check the specific error message. If the message matches the error of the object not existing the client knows the object doesn't exist.

Reply:
The technique you suggest doesn't scale.  It's basically checking strings--really brittle. To detect errors properly you need something that's backed by the type system, e.g., an enum that you can reliably detect in client code.  The compiler will guarantee you are looking for something meaninful.

Reply:
Thats a good point. I will change that. I will also make an issue for other functions that involve getting an entity be changed to this format